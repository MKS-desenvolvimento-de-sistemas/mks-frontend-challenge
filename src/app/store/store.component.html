<div class="store">
    <h1>Store</h1>
    <div *ngIf="isLoading">Loading...</div>
    <app-product-card *ngFor="let product of products" [product]="product" (addToCartEvent)="addToCart($event)"></app-product-card>
  </div>
  