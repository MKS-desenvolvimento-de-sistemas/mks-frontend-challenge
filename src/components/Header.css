.header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color: #0f52ba;
    padding: 1rem;
  }
  
  .logo {
    font-size: 1.5rem;
    color: #fff;
  }
  
  .sistemas {
    font-weight: 300;
  }
  
  .cart-icon {
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: #fff;
    width: 90px;
    height: 45px;
    border-radius: 8px;
  }
  
  .cart-icon img {
    width: 24px;
    height: 24px;
    margin-right: 0.5rem;
  }
  
  .cart-icon span {
    font-size: 1.2rem;
  }
  