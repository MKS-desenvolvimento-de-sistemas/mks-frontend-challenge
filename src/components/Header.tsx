import React from "react";
import Cart from "./Cart";

class Header extends React.Component {
  render() {
    return <header><p>MKS Sistem</p><Cart /></header>;
  }
}

export default Header;