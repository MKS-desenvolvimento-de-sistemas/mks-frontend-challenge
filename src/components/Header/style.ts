/* eslint-disable @typescript-eslint/naming-convention */
import styled from 'styled-components';

export const Container = styled.div`
  max-width: 1216px;
  margin: 0 auto;
  height: 101px;
  display: flex;
  padding-left: 8px;
  padding-right: 8px;
  justify-content: space-between;
  align-items: center;
`;

export const HeaderHome = styled.header`
  background: #0F52BA;
`;

export const CartButton = styled.header`
  align-items: center;
  background: white;
  border-radius: 8px;
  cursor: pointer;
  display: flex;
  justify-content: center;
  height: 45px;
  width: 90px;
  
  img {
    margin-right: 16px;
  }

  span {
    font-size: 18px;
    font-weight: 700;
  }
`;
