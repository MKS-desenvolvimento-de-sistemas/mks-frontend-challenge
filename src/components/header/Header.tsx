import IconCart from "../../../assets/shopping-cart.png";

import Image from "next/image";

import * as S from "./styles";

import { useState } from "react";

import FinishShopping from "../finishShopping/FinishShopping";
import { useShopping } from "@/context/shopping/useShopping";

const Header = () => {
  const [open, setOpen] = useState(false); // Controla a abertura e o fechamento da aba de compras

  const openShopping = () => setOpen(true);
  const closeShopping = () => setOpen(false);

  const { shopping } = useShopping();

  return (
    <S.Container>
      <S.Logo>
        <S.Title>MKS</S.Title>
        <S.SubTitle>Sistemas</S.SubTitle>
      </S.Logo>

      <S.Shopping onClick={openShopping}>
        <Image src={IconCart} alt="shopping cart" width={19} height={18} />

        <S.QuantityOfProducts>{shopping.length}</S.QuantityOfProducts>
      </S.Shopping>
      {open && <FinishShopping closeShopping={closeShopping} />}
    </S.Container>
  );
};

export default Header;
