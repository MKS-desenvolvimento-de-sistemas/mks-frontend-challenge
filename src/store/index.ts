import { createSlice } from "@reduxjs/toolkit";
import { configureStore } from "@reduxjs/toolkit";
import { ICartItem } from "../pages/Cart";
const initialState = { items: [] } as any;

const cartSlice = createSlice({
  name: "carrinho",
  initialState,
  reducers: {
    addToCart(state, action) {
      state.items.push(action.payload);
    },

    addAdditional(state, action) {
      if (!state.items.some((item) => item.id === action.payload.id)) return;
      const item = state.items.find((item) => item.id === action.payload.id);

      state.items.push(item);
    },
    decreaseItem(state, action) {
      const remainingItems = state.items.filter(
        (item) => item.id === action.payload.id
      );

      if (remainingItems.length === 1) return;

      const itemIndex = state.items.findIndex(
        (el) => el.id === action.payload.id
      );

      state.items.splice(itemIndex, 1);
    },
    deleteItem(state, action) {
      const arr = state.items.filter((item) => item.id !== action.payload.id);
      state.items = arr;
      // do {
      //   const itemIndex = state.items.findIndex(
      //     (el) => el.id === action.payload.id
      //   );

      //   state.items.splice(itemIndex, 1);
      // } while (state.items.some((item) => item.id === action.payload.id));
    },
    deleteAllItems(state) {
      state.items = [];
    },
  },
});

const store = configureStore({
  reducer: cartSlice.reducer,
});

export const cartActions = cartSlice.actions;

export default store;
