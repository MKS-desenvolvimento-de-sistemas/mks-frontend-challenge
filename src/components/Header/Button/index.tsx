import React, { ButtonHTMLAttributes } from "react";
import { IconBaseProps } from "react-icons";
import { ButtonContainer } from "./style";

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  icon?: React.ComponentType<IconBaseProps>;
}

export const Button: React.FC<ButtonProps> = ({
  icon: Icon,
  children,
  ...rest
}) => {
  return (
    <ButtonContainer>
      {Icon && <Icon color="#000" size={17} />}
      <p>{children}</p>
    </ButtonContainer>
  );
};
