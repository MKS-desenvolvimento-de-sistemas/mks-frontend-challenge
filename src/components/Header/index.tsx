import React from 'react';
import {Container, HeaderHome, CartButton} from './style';
import Logo from '../../img/logo.svg';
import Cart from '../../img/cart.png';

const Header = () => (
	<HeaderHome>
		<Container>
			<div>
				<img src={Logo} alt='Logo MKS sistemas' />
			</div>
			<CartButton>
				<img src={Cart} alt='Carrinho de compra' />
				<span>0</span>
			</CartButton>
		</Container>
	</HeaderHome>
);

export default Header;
