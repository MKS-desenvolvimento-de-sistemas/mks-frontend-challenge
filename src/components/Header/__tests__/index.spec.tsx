import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";
import Header from "../";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from "../../../redux/";

describe("Header Component", () => {
  it("should render the Link correctly", () => {
    const openCart = jest.fn();
    render(
      <Provider store={store}>
        <Header openCart={openCart} />
      </Provider>
    );

    const header = screen.getByTestId("header");

    expect(header).toHaveTextContent("MKS");
    expect(header).toHaveTextContent("Sistemas");
  });

  it("should render the Theme Button correctly", () => {
    const openCart = jest.fn();
    render(
      <Provider store={store}>
        <Header openCart={openCart} />
      </Provider>
    );

    const themeButton = screen.getByTestId("theme-button");
    expect(themeButton.getAttribute("type")).toEqual("button");
    expect(themeButton.getAttribute("title")).toEqual("Ativar Modo Escuro");

    fireEvent.click(themeButton);

    expect(themeButton.getAttribute("title")).toEqual("Ativar Modo Claro");
  });

  it("should render the Cart Button correctly", () => {
    const openCart = jest.fn();
    render(
      <Provider store={store}>
        <Header openCart={openCart} />
      </Provider>
    );

    const cartButton = screen.getByTestId("cart-button");
    expect(cartButton.getAttribute("type")).toEqual("button");
    expect(cartButton.getAttribute("title")).toEqual("Abrir Carrinho");
  });

  it("should match the snapshot", () => {
    const openCart = jest.fn();
    const tree = renderer
      .create(
        <Provider store={store}>
          <Header openCart={openCart} />
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
