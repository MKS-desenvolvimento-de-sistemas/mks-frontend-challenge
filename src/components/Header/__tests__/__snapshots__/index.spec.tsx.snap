// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match the snapshot 1`] = `
<header
  className="sc-bczRLJ fdqXQM"
  data-testid="header"
>
  <a
    data-testid="link"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
    target="_self"
    title="MKS Sistemas"
  >
    MKS
    <span>
      Sistemas
    </span>
  </a>
  <button
    className="sc-dkzDqf hgIJhX"
    data-testid="theme-button"
    onClick={[Function]}
    title="Ativar Modo Claro"
    type="button"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"
      />
    </svg>
  </button>
  <button
    className="sc-gsnTZi fhyRXI"
    data-testid="cart-button"
    onClick={[MockFunction]}
    title="Abrir Carrinho"
    type="button"
  >
    <i>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="176"
          cy="416"
          r="32"
        />
        <circle
          cx="400"
          cy="416"
          r="32"
        />
        <path
          d="M456.8 120.78a23.92 23.92 0 00-18.56-8.78H133.89l-6.13-34.78A16 16 0 00112 64H48a16 16 0 000 32h50.58l45.66 258.78A16 16 0 00160 368h256a16 16 0 000-32H173.42l-5.64-32h241.66A24.07 24.07 0 00433 284.71l28.8-144a24 24 0 00-5-19.93z"
        />
      </svg>
    </i>
    0
  </button>
</header>
`;
